dallinger/command_line.py

```diff
@@ -283,7 +283,10 @@ def setup_experiment(debug=True, verbose=False, app=None, exp_config=None):
 
     # Verify that the Postgres server is running.
     try:
-        psycopg2.connect(database="x", user="postgres", password="nada")
+#        psycopg2.connect(database="x", user="postgres", password="nada")
+        db_url_default = "postgresql://dallinger:dallinger@localhost/dallinger"
```

Comment:
I don't understand the original code (was the `connect` call generally working despite the nonsense password?), so it's hard for me to assess the change. Will using the variant of the URL @alecpm suggests implicitly use a different password, or is a password not required for this call based on our setup instructions?

Reply:
I didn't have a chance (time) to try and understand what the original code was for.  I just know that this modification, makes things work with the docker-compose setup.
What I am hoping for is to have some conditional for the original line of code (if necessary) to work alongside the docker-compose adaptation, and I was hoping someone who is more familiar with the code base would perhaps have more insight?

Reply:
Something like:
```
if Dallinger running not in docker:
  psycopg2.connect(database="x", user="postgres", password="nada")
elif dallinger running in docker
        db_url_default = "postgresql://dallinger:dallinger@localhost/dallinger"
        db_url = os.environ.get("DATABASE_URL", db_url_default)
        psycopg2.connect(dsn=db_url)
```
so that they can coexist together somehow.

Reply:
Ideally the new, URL-based call will work for both contexts (Docker and "normal"), and we won't need a condition. Looks like the tests pass already, so hopefully we'll be able to just replace the manner of calling `connect()`.

Reply:
In fact Dallinger in docker doesn't need this check for its own sake, since in docker-compose there is a health check, so dallinger waits till postgres is in a healthy state before trying to connect to it.
However this original code fails and gives a false positive to dallinger in the docker situation.
```
  psycopg2.connect(database="x", user="postgres", password="nada")
```
I don't know if this code works in the stand-alone non-docker dallinger mode, but I guess it should since its in the codebase right?